[CANCELLED] 改进 schedule.js 逻辑等等

dearrrfish opened the pull request:
- 修改 config.js 中全局变量赋值，声明 `self` scope，并提供默认值。这样便可以在 wrangler.toml 中自定义并省略可选值。
- 增加若干变量，具体可见 wrangler.example.toml 中注释
- 增加 webpack.config.dev.js，关闭 minify，方便在 Cloudflare Editor 中修改调试
- ~identify() 改进~ WIP
- ~schedule.js 中逻辑改进~ WIP

dearrrfish committed:
feat($schedule): improved schedule task logic

pureink commented:
首先非常感谢贡献，但对于identify函数的修改我测试时发现一些问题（主要是因为我没有为该项目添加测试导致的）。比如 https://github.com/rauchg.atom 或者 https://blog.cloudflare.com/rss/ 并不能成功区分。我还在准备考研并没有很仔细阅读您的修改，能否说明一下改进为按照guid > link > title 的优先顺序选取值方式带来的好处？仅用title检测更新我认为是足够的。
其次对于您提到的feeds don't sort items by `pubDate` ，feeds don't provide correct `pubDate`和feeds don't have `pubDate`我认为属于一类问题，即使用其他方式找到了新出现的item但他们在时间上并不一定是最新的，我认为通知并不适用于榜单之类的内容，他们没有体现出更“新”，对于频繁更新变化的榜单等内容，可能会造成大量的通知。但这毕竟只是我的个人看法，不代表所有用户，不过还是希望您能提供我一些这样做的理由。
再次感谢您的贡献！

dearrrfish commented:
关于 identify 函数我也还在测试改进，比如你的第一个例子是 entry 而不是 item tag，regex 没有覆盖到所以失效。的确还需要更多的 match 条件来覆盖其他常见的 schema。

guid > link > title 还是考虑到 item 标记的唯一性，不过如果只是作为一个简单的通知工具不考虑 edge caeses， title 大部分情况的确够用。

关于 pubDate 我的做法可能太偏激，测试下来也对某些 feed 也带来重复通知的副作用，正在考虑修正的方法，比如排序。

榜单类或合集类 RSS 我个人使用频率挺高的，可能加入 feed 自定义选项可以解决这类问题（更新频率、排序、自定义 updateID 等）

dearrrfish committed:
feat($schedule): updated identify func seeking for `lastBuildDate`;
added new field `lastUpdateItem` to keep last sent item; sorted items by
pubDate for non-ranking feeds

pureink commented:
lastBuildDate并不适用于所有的订阅源，除去rsshub的内容，部分博客的rss并不是静态文件，举个例子 https://blog.spencerwoo.com/feed 每次访问时都会发生变化，但事实上并没有产生更新。我暂时还是持有identify函数没有改动必要的观点，使用title还有一个好处是可以提供给外部使用
![](https://user-images.githubusercontent.com/44235276/126298401-3370afa4-878d-4243-a587-302e60a19fa3.png)

dearrrfish commented:
同意，是过于个人向了。那我先关闭这个 PR Draft。考研加油！
